token-lending/program/src/processor.rs

```diff
@@ -239,30 +239,36 @@ fn process_set_lending_market_owner_and_config(
 ) -> ProgramResult {
     let account_info_iter = &mut accounts.iter();
     let lending_market_info = next_account_info(account_info_iter)?;
-    let lending_market_owner_info = next_account_info(account_info_iter)?;
+    let signer_info = next_account_info(account_info_iter)?;
 
     let mut lending_market = LendingMarket::unpack(&lending_market_info.data.borrow())?;
     if lending_market_info.owner != program_id {
         msg!("Lending market provided is not owned by the lending program");
         return Err(LendingError::InvalidAccountOwner.into());
     }
-    if &lending_market.owner != lending_market_owner_info.key {
-        msg!("Lending market owner does not match the lending market owner provided");
-        return Err(LendingError::InvalidMarketOwner.into());
-    }
-    if !lending_market_owner_info.is_signer {
-        msg!("Lending market owner provided must be a signer");
-        return Err(LendingError::InvalidSigner.into());
-    }
 
-    lending_market.owner = new_owner;
-    lending_market.risk_authority = risk_authority;
+    if signer_info.key == &lending_market.owner {
+        lending_market.owner = new_owner;
+        lending_market.risk_authority = risk_authority;
 
-    if rate_limiter_config != lending_market.rate_limiter.config {
-        lending_market.rate_limiter = RateLimiter::new(rate_limiter_config, Clock::get()?.slot);
+        if rate_limiter_config != lending_market.rate_limiter.config {
+            lending_market.rate_limiter = RateLimiter::new(rate_limiter_config, Clock::get()?.slot);
+        }
+
+        lending_market.whitelisted_liquidator = whitelisted_liquidator;
+    } else if signer_info.key == &lending_market.risk_authority {
+        if rate_limiter_config != lending_market.rate_limiter.config {
+            lending_market.rate_limiter = RateLimiter::new(rate_limiter_config, Clock::get()?.slot);
+        }
+    } else {
+        msg!("Signer must be the lending market owner or risk authority");
+        return Err(LendingError::InvalidMarketOwner.into());
     }
 
-    lending_market.whitelisted_liquidator = whitelisted_liquidator;
+    if !signer_info.is_signer {
```

Comment:
slight preference for this to be near the top of constraints rather than bottom, and rename variable to like market_change_authority (just to not mix up to concept of it being a signer)